account_invoice_split/wizard/account_invoice_split.py

```diff
@@ -0,0 +1,160 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015-2017 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models, fields, api, exceptions, _
+from odoo.exceptions import UserError
+
+import odoo.addons.decimal_precision as dp
+
+FIELD_MAPPING = [
+    ('id', 'origin_invoice_line_id'),
+    ('product_id', 'product_id'),
+    ('name', 'name'),
+    ('quantity', 'total_quantity')]
+
+
+class AccountInvoiceSplit(models.TransientModel):
+    _name = 'account.invoice.split'
+
+    line_ids = fields.One2many(
+        comodel_name='account.invoice.split.line', inverse_name='wizard_id',
+        string='Invoice lines to Split')
+
+    @api.model
+    def _get_invoice_id_from_context(self):
+        active_ids = self.env.context.get('active_ids')
+        if len(active_ids) != 1:
+            raise exceptions.UserError(
+                _("You can call this action only for a single invoice."))
+        return active_ids[0]
+
+    @api.model
+    def _get_invoice_from_context(self):
+        invoice_id = self._get_invoice_id_from_context()
+        invoice = self.env['account.invoice'].browse(invoice_id)
+        self._check_invoice(invoice)
+        return invoice
+
+    @api.model
+    def _check_invoice(self, invoice):
+        if invoice.state != 'draft':
+            raise UserError(
+                _('Invoice must be draft, not %s') % (invoice.state,))
+
+    @api.model
+    def default_get(self, fields_list):
+        res = super(AccountInvoiceSplit, self).default_get(fields_list)
+        invoice = self._get_invoice_from_context()
+        split_lines = []
+        for line in invoice.invoice_line_ids:
+            current_line_values = {
+                'origin_invoice_line_id': line.id,
+                'product_id': line.product_id.id,
+                'name': line.name,
+                'total_quantity': line.quantity,
+            }
+            split_lines.append((0, 0, current_line_values))
+        res['line_ids'] = split_lines
```

Comment:
@benwillig (Nitpicking) Override of `default_get` could be avoided by doing this in a default method of `line_ids` field.

Reply:
I prefer to have only one method to initialize wizard values.